images/kindnetd/cmd/kindnetd/cni.go

```diff
@@ -28,27 +29,49 @@ import (
 	"k8s.io/utils/net"
 )
 
+//sensibleMTU value to fall back on - if we're not able to configure from eth0
+const defaultMtu = 1480
+
 /* cni config management */
 
 // CNIConfigInputs is supplied to the CNI config template
 type CNIConfigInputs struct {
 	PodCIDR      string
 	DefaultRoute string
+	Mtu          int
 }
 
 // ComputeCNIConfigInputs computes the template inputs for CNIConfigWriter
 func ComputeCNIConfigInputs(node corev1.Node) CNIConfigInputs {
 	podCIDR := node.Spec.PodCIDR
 	defaultRoute := "0.0.0.0/0"
+	mtu := computeBridgeMTU()
+
 	if net.IsIPv6CIDRString(podCIDR) {
 		defaultRoute = "::/0"
 	}
 	return CNIConfigInputs{
 		PodCIDR:      podCIDR,
 		DefaultRoute: defaultRoute,
+		Mtu:          mtu,
 	}
 }
 
+//computeBridgeMTU finds the mtu for the eth0 interface
+//otherwise it defults to a sensible value - 1480
+func computeBridgeMTU() int {
+	interfaces, err := stdnet.Interfaces()
+	if err != nil {
+		return defaultMtu
+	}
+	for _, inter := range interfaces {
```

Comment:
let's make it smarter, 
we can default to eth0 but if there are any other interface with an ip we can use it
```go
		for _, i := range ifaces {
			addrs, err := i.Addrs()
			if err != nil {
				continue
			}

			for _, addr := range addrs {
				var ip net.IP
				switch v := addr.(type) {
				case *net.IPNet:
					ip = v.IP
				case *net.IPAddr:
					ip = v.IP
				}

				if ip.IsGlobalUnicast() {
					return i.MTU
				}
```

Reply:
Dismiss this comment, let's go with current approach, this can be done Ina follow up, no need to complicate it more unnecessarily.
Sorry